saleor/order/utils.py

```diff
@@ -236,11 +231,8 @@ def update_order_prices(order: Order, manager: "PluginsManager", tax_included: b
     update_taxes_for_order_lines(order.lines.all(), order, manager, tax_included)
 
     if order.shipping_method:
-        shipping_price = manager.calculate_order_shipping(order)
-        order.shipping_price = shipping_price
-        order.shipping_tax_rate = manager.get_order_shipping_tax_rate(
-            order, shipping_price
-        )
+        order.shipping_price = calculations.order_shipping(order, manager)
```

Comment:
If I am not wrong, you're calling  `.save()` twice for these values. Inside a `fetch_if_expired` and here. Keep an eye on such things as we can't allow for duplicating DB requests as it has huge impacts on performance overall.